flowmachine/flowmachine/features/subscriber/distance_series.py

```diff
@@ -0,0 +1,140 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Per subscriber time series of distances from some reference location.
+"""
+from typing import List, Optional
+
+from flowmachine.features.spatial import DistanceMatrix
+from .metaclasses import SubscriberFeature
+from ..utilities.subscriber_locations import SubscriberLocations, BaseLocation
+
+
+valid_stats = {"sum", "avg", "max", "min", "median", "stddev", "variance"}
+valid_time_buckets = ["second", "minute", "hour", "day", "month", "year", "century"]
+
+
+class DistanceSeries(SubscriberFeature):
+    """
+    Per subscriber time series of distance in meters from some reference location.
+
+    Parameters
+    ----------
+    subscriber_locations : SubscriberLocations
+        A subscriber locations query with a lon-lat spatial unit to build the distance series against.
+    reference_location : BaseLocation or None, default None
+        The set of home locations from which to calculate distance at each sighting.
+        If not given then distances will be from (0, 0).
+    statistic : str
+        the statistic to calculate one of 'sum', 'avg', 'max', 'min', 
+        'median', 'stddev' or 'variance'
+    time_bucket : {"second", "minute", "hour", "day", "month", "year", "century"}, default "day"
+        Time bucket to calculate the statistic over.
```

Comment:
When I tried this out I was slightly confused that for `time_bucket="month"/"year"` it would still return single dates in the `datetime` column. What do these dates refer to? Are they the first date in the time bucket? Would be good to clarify this in the docstring.